docs/types/operators.rst

```diff
@@ -1,7 +1,34 @@
-.. index:: assignment, ! delete, lvalue
+.. index:: operator
 
-Operators Involving LValues
-===========================
+Operators
+=========
+
+Binary operators can be applied even if the two operands do not have the same type.
+For example, you can compute ``y = x + z``, where ``x`` is a ``uint8`` and ``z`` has
+the type ``int32``. In these cases, the following mechanism will be used to determine
+the type in which the operation is computed (this is important in case of overflow)
+and the type of the operator result:
+
+1. If the type of the right operand can be implicitly converted to the type of the left
+   operand, use the type of the left operand,
+2. if the type of the left operand can be implicitly converted to the type of the right
+   operand, use the type of the right operand,
```

Comment:
Maybe we should mention that operator `=` is a special case and the second rule does not apply to it?

Reply:
Sure, `=` does not even perform an operation.

Reply:
I mean, boolean operators are also special. Maybe limit this to "arithmetic and bitwise operators"?

Reply:
Sounds good.

Reply:
> Sure, `=` does not even perform an operation.

You can use it in an expression though and it returns a value so technically it's still a binary operator, right?